packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -154,279 +149,6 @@ describe('anchor', () => {
       await ceramic2.close()
     })
 
-    test('commit history and atCommit', async () => {
-      const stream = await TileDocument.create<any>(ceramic, INITIAL_CONTENT)
-      stream.subscribe()
-      const streamState = await ceramic.repository.load(stream.id, {})
-
-      const commit0 = stream.allCommitIds[0]
-      expect(stream.commitId).toEqual(commit0)
-      expect(commit0.equals(CommitID.make(streamState.id, streamState.id.cid))).toBeTruthy()
-
-      await TestUtils.anchorUpdate(ceramic, stream)
-      expect(stream.allCommitIds.length).toEqual(2)
-      expect(stream.anchorCommitIds.length).toEqual(1)
-      const commit1 = stream.allCommitIds[1]
-      expect(commit1.equals(commit0)).toBeFalsy()
-      expect(commit1).toEqual(stream.commitId)
-      expect(commit1).toEqual(stream.anchorCommitIds[0])
-
-      const newContent = { abc: 321, def: 456, gh: 987 }
-      const updateRec = await stream.makeCommit(ceramic, newContent)
-      await ceramic.repository.applyCommit(streamState.id, updateRec, {
-        anchor: true,
-        publish: false,
-      })
-      expect(stream.allCommitIds.length).toEqual(3)
-      expect(stream.anchorCommitIds.length).toEqual(1)
-      const commit2 = stream.allCommitIds[2]
-      expect(commit2.equals(commit1)).toBeFalsy()
-      expect(commit2).toEqual(stream.commitId)
-
-      await TestUtils.anchorUpdate(ceramic, stream)
-      expect(stream.allCommitIds.length).toEqual(4)
-      expect(stream.anchorCommitIds.length).toEqual(2)
-      const commit3 = stream.allCommitIds[3]
-      expect(commit3.equals(commit2)).toBeFalsy()
-      expect(commit3).toEqual(stream.commitId)
-      expect(commit3).toEqual(stream.anchorCommitIds[1])
-      expect(stream.content).toEqual(newContent)
-      expect(stream.state.signature).toEqual(SignatureStatus.SIGNED)
-      expect(stream.state.anchorStatus).not.toEqual(AnchorStatus.NOT_REQUESTED)
-      expect(stream.state.log.length).toEqual(4)
-
-      // Apply a final commit that does not get anchored
-      const finalContent = { foo: 'bar' }
-      const updateRec2 = await stream.makeCommit(ceramic, finalContent)
-      await ceramic.repository.applyCommit(streamState.id, updateRec2, {
-        anchor: true,
-        publish: false,
-      })
-
-      expect(stream.allCommitIds.length).toEqual(5)
-      expect(stream.anchorCommitIds.length).toEqual(2)
-      const commit4 = stream.allCommitIds[4]
-      expect(commit4.equals(commit3)).toBeFalsy()
-      expect(commit4).toEqual(stream.commitId)
-      expect(commit4.equals(stream.anchorCommitIds[1])).toBeFalsy()
-      expect(stream.state.log.length).toEqual(5)
-
-      await TestUtils.waitForAnchor(stream, 3000)
-
-      // Correctly check out a specific commit
-      const streamStateOriginal = cloneDeep(streamState.state)
-      const streamV0 = await ceramic.repository.stateManager.atCommit(streamState, commit0)
```

Comment:
All the tests removed from `state-manager.test.ts` have been moved to `repository.test.ts`.  They're almost all identical except that calls to `repository.stateManager.atCommit` have been replaced with calls to `repository.loadAtCommit`.  By updating the tests to work at a higher abstraction level, I'll be able to swap out the implementation of `repository.loadAtCommit` to one based on the `StreamLoader` without losing any test coverage.  After `repository.loadAtCommit` has been updated to use the `StreamLoader`, `stateManager.atCommit` can be deleted entirely.  I actually hope that by the end of this refactoring the entire `StateManager` has been removed, with its functionality either folded up into `Repository` or relocated into more special-purpose sub-modules (such as the `StreamLoader` and soon-to-come `StreamUpdater`).